serialport.go

```diff
@@ -92,61 +92,63 @@ type SpPortMessageRaw struct {
 	D []byte // the data, i.e. G0 X0 Y0
 }
 
-func (p *serport) reader() {
+func (p *serport) reader(buftype string) {
 
 	//var buf bytes.Buffer
-	ch := make([]byte, 1024)
 	timeCheckOpen := time.Now()
 	var buffered_ch bytes.Buffer
 
 	for {
-
+		ch := make([]byte, 1024) //a new array of bytes is initilized everytime because we pass it (as a pointer) in a channel, it can be improved
```

Comment:
I do not like this solution, it should be improved, I think that causes the CPU to work a lot

Reply:
Instantiate `ch` outside the loop and rename it to `serialBuffer`, then change these lines to:

```
		n, err := p.portIo.Read(serialBuffer)
		bufferPart := serialBuffer[:n]
```

You'll have to change all the occurences of `ch` below to use `bufferPart`.